From c511b49bf697e05df4b0987d0fd81e3329f64ce6 Mon Sep 17 00:00:00 2001
From: Takashi Iwai <tiwai@suse.de>
Date: Fri, 28 Jan 2022 17:46:45 +0100
Subject: [PATCH] alsamixer: Revert has_mouse() check

has_mouse() function of ncurses doesn't seem working reliably.
Revert the previous change for addressing the regressions.

Fixes: 31820c5f239f ("alsamixer: Check the availability of mouse")
BugLink: https://github.com/alsa-project/alsa-utils/issues/139
Signed-off-by: Takashi Iwai <tiwai@suse.de>
---
 alsamixer/mainloop.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/alsamixer/mainloop.c b/alsamixer/mainloop.c
index cdbe28e390fd..bf3f70defbaa 100644
--- a/alsamixer/mainloop.c
+++ b/alsamixer/mainloop.c
@@ -50,8 +50,9 @@ void initialize_curses(bool use_color, bool use_mouse)
 #endif
 	window_size_changed(); /* update screen_lines/cols */
 	init_colors(use_color);
-	if (use_mouse && has_mouse())
+	if (use_mouse)
 		mousemask(ALL_MOUSE_EVENTS, NULL);
+
 	snd_lib_error_set_handler(black_hole_error_handler);
 }
 
-- 
2.31.1

